eth2/utils/eth2_hashing/src/lib.rs

```diff
@@ -27,10 +27,18 @@ pub fn hash(input: &[u8]) -> Vec<u8> {
 }
 
 /// Compute the hash of two slices concatenated.
-pub fn hash_concat(h1: &[u8], h2: &[u8]) -> Vec<u8> {
-    let mut vec1 = h1.to_vec();
-    vec1.extend_from_slice(h2);
-    hash(&vec1)
+///
+/// # Panics
+///
+/// Will panic if either `h1` or `h2` are not 32 bytes in length.
+pub fn hash32_concat(h1: &[u8], h2: &[u8]) -> [u8; 32] {
+    let mut preimage = [0; 64];
+    preimage[0..32].copy_from_slice(h1);
+    preimage[32..64].copy_from_slice(h2);
+
+    let mut output = [0; 32];
+    output[..].copy_from_slice(&hash(&preimage));
+    output
```

Comment:
Have you tried [`ring::digest::Context`](https://docs.rs/ring/0.16.9/ring/digest/struct.Context.html) here? It's probably no quicker, but it avoids the panic

Reply:
I tested it, no noticeable improvement but it does avoid a heap allocation so I left it in :)

Reply:
Cool cool